### PR TITLE
fix #99

### LINKED ISSUE #99: extend soc ctrl to have value of the target frequency saved

### PR DESCRIPTION
**Pull request by davideschiavone**

### CODE REVIEW

File: sw/device/lib/runtime/syscalls.c [hunk 1]
```diff
@@ -239,10 +239,13 @@ ssize_t _write(int file, const void *ptr, size_t len)
...
     }
 
+    soc_ctrl_t soc_ctrl;
+    soc_ctrl.base_addr = mmio_region_from_addr((uintptr_t)SOC_CTRL_START_ADDRESS);
+
     uart_t uart;
     uart.base_addr   = mmio_region_from_addr((uintptr_t)UART_START_ADDRESS);
-    uart.baudrate    = 256000;
-    uart.clk_freq_hz = REFERENCE_CLOCK_KHz*1000;
+    uart.baudrate    = UART_BOUDRATE;
```
Comment (benoitdenkinger) [inline sw/device/lib/runtime/syscalls.c:247]: UART_BAUDRATE

File: sw/device/target/pynq-z2/x-heep.h [hunk 2]
```diff
@@ -12,7 +12,8 @@ extern "C" {
 #endif  // __cplusplus
 
 
-#define REFERENCE_CLOCK_KHz 20000
+#define REFERENCE_CLOCK_Hz 20*1000*1000
+#define UART_BOUDRATE 115200
```
Comment (benoitdenkinger) [inline sw/device/target/pynq-z2/x-heep.h:16]: Ah ok I found it. Change to UART_BAUDRATE

File: sw/device/target/sim/x-heep.h [hunk 3]
```diff
@@ -12,7 +12,8 @@ extern "C" {
 #endif  // __cplusplus
 
 
-#define REFERENCE_CLOCK_KHz 100000
+#define REFERENCE_CLOCK_Hz 100*1000*1000
+#define UART_BOUDRATE 256000
```
Comment (benoitdenkinger) [inline sw/device/target/sim/x-heep.h:16]: Same here UART_BAUDRATE

Comment (davideschiavone) [reply sw/device/target/sim/x-heep.h]: typos are my favourite errors <3

**Pull request merged**